annotation/tests/test_distribution.py

```diff
@@ -1516,3 +1523,139 @@ def test_prepare_response():
             )
             == expected_output
         )
+
+
+@pytest.fixture
+def mock_session_and_tasks():
+    task_list = [
+        ManualAnnotationTask(
+            id=0,
+            status=TaskStatusEnumSchema.finished,
+            is_validation=True,
+            file_id=0,
+            pages=[1, 2, 3, 4],
+        ),
+        ManualAnnotationTask(
+            id=1,
+            status=TaskStatusEnumSchema.finished,
+            is_validation=False,
+            file_id=1,
+            pages=list(range(20)),
+        ),
+    ]
+
+    mock_db = Mock(spec=Session, flush=Mock(), rollback=Mock())
+    mock_db.query.return_value.filter.return_value.all.return_value = task_list
+
+    return mock_db, task_list
+
+
+@pytest.fixture
+def patch_redistribute_dependencies():
+    with patch(
+        "annotation.distribution.main.get_pages_in_work", return_value=None
+    ) as mock_get_pages, patch(
+        "annotation.distribution.main.distribute", return_value=None
+    ) as mock_distribute, patch(
+        "annotation.distribution.main.set_task_statuses"
+    ) as mock_set_task_statuses:
+        yield mock_get_pages, mock_distribute, mock_set_task_statuses
+
+
+@pytest.mark.parametrize(
+    (
+        "job_files",
+        "job_status",
+    ),
+    (
+        (
+            [
+                File(file_id=0, pages_number=10),
+                File(file_id=1, pages_number=20),
+            ],
+            JobStatusEnumSchema.in_progress,
+        ),
+        ([], JobStatusEnumSchema.finished),
+    ),
+)
+def test_redistribute(
+    mock_session_and_tasks: Mock,
+    patch_redistribute_dependencies: Mock,
+    job_files: List[File],
+    job_status: JobStatusEnumSchema,
+):
+    job_id = 0
+    callback_url = "url"
+    job = Job(
+        job_id=job_id,
+        files=job_files,
+        status=job_status,
+        callback_url=callback_url,
+    )
+
+    mock_db, task_list = mock_session_and_tasks
+    (
+        mock_get_pages,
+        mock_distribute,
+        mock_set_task_statuses,
+    ) = patch_redistribute_dependencies
+
+    def delete_tasks_side_effect(_, tasks_to_delete):
+        nonlocal task_list
+        task_list = [task for task in task_list if task not in tasks_to_delete]
```

Comment:
I dont see the point of this side effect. Remove it and change the patch to make it simplier.

Reply:
My idea was to have a mock database with tasks. This function `delete_tasks` is supposed to delete tasks from the database, but i thought i would manipulate the list of tasks i pass directly. I'm not sure how to make this idea simpler

Reply:
But why do you need that? you dont check for values it changes really, you check if it as called with certain params, that's all. You dont need such side effect

Reply:
Yeah, I get it now. I mocked the function with a side effect but only now realize I didn't need any side effect. I'll remove that